Gem/Code/Source/Components/NetworkWeaponsComponent.h

```diff
@@ -68,6 +70,7 @@ namespace MultiplayerSample
     class NetworkWeaponsComponentController
         : public NetworkWeaponsComponentControllerBase
         , private StartingPointInput::InputEventNotificationBus::MultiHandler
+        , private AZ::TickBus::Handler
```

Comment:
It might be nicer to use scheduled events for these sorts of polling operations?  AZ::ScheduledEvent.  This way if you spawn a few million of these critters there will be some amount of batching and prioritization so we can maintain a consistent tick rate rather than forcibly linearly visiting every registered component.

Reply:
Haven't heard of that till now, will take a look thanks!